src/main/telemetry/msp_shared.c

```diff
@@ -164,6 +178,33 @@ bool handleMspFrame(uint8_t *frameStart, uint8_t *frameEnd)
     return true;
 }
 
+void bufferMspFrame(uint8_t *frameStart, uint8_t *frameEnd)
+{
+    const uint8_t mspFrameLength = frameEnd-frameStart;
+
+    if (rbufWriteBytesRemaining(mspRxRingBuffer.ring)>=mspFrameLength) {
+        rbufWriteU8(mspRxRingBuffer.ring, mspFrameLength);
+        rbufWriteData(mspRxRingBuffer.ring, frameStart, mspFrameLength);
+        mspRxRingBuffer.frames++;
+    }
+}
+
+bool handleMspBufferFrame(uint8_t payloadSize, mspResponseFnPtr responseFn)
+{
+    static bool requestHandled = false;
+
+    while (mspRxRingBuffer.frames>0) {
```

Comment:
`mspRxRingBuffer.frames` is used for synchronization into ISR - some mechanism to prevent race conditions should be used ..

Reply:
this will be a data driven condition in a coming update that doesn't require a counter.